### PR TITLE
Feature/plcn 128 testing hubspot services

### PR DESCRIPTION
**Pull request by Nicolai-Bak**

### CODE REVIEW

File: Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HashComputer/SHA256HashComputerTests.cs [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+		_uut = new();
+	}
+
+	[Fact]
+	public void ComputeHash()
+	{
+		/// Arrange
+		/// HubSpot Example
+		string clientSecret = "yyyyyyyy-yyyy-yyyy-yyyy-yyyyyyyyyyyy";
+		string body = "[{\"eventId\":1,\"subscriptionId\":12345,\"portalId\":62515,\"occurredAt\":1564113600000,\"subscriptionType\":\"contact.creation\",\"attemptNumber\":0,\"objectId\":123,\"changeSource\":\"CRM\",\"changeFlag\":\"NEW\",\"appId\":54321}]";
```
Comment (llh-itm) [inline Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HashComputer/SHA256HashComputerTests.cs:21]: having this as one long string makes it hard to read and to change later if needed. i would recommend splitting it into new lines or reformatting like this:
```C#
string body = @"[{
	""eventId"":1,
	""subscriptionId"":12345,
	""portalId"":62515,	
	""occurredAt"":1564113600000,
	""subscriptionType"":""contact.creation"",
	""attemptNumber"":0,
	""objectId"":123,
	""changeSource"":""CRM"",
	""changeFlag"":""NEW"",
	""appId"":54321
}]";
```

File: Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HubSpotValidationFilterTests.cs [hunk 2]
```diff
@@ -0,0 +1,578 @@
...
+		/// Assert
+		Assert.Null(exception);
+
+		_loggerMock.Verify(x =>
+			x.Log(
+				LogLevel.Warning,
+				It.IsAny<EventId>(),
+				It.Is<It.IsAnyType>((v, t) => true),
+				It.IsAny<Exception>(),
+				It.Is<Func<It.IsAnyType, Exception, string>>((v, t) => true)),
```
Comment (llh-itm) [inline Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HubSpotValidationFilterTests.cs:98]: add ?-operator at the end of Exception to remove null warning

File: Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HubSpotValidationFilterTests.cs [hunk 3]
```diff
@@ -0,0 +1,578 @@
...
+			Times.Once());
+
+		Assert.Equal(
+			typeof(BadRequestObjectResult),
+			resourceExecutingContext.Result?.GetType());
+
+		Assert.Equal(
+			"{ message = Unable to validate signature }",
+			((BadRequestObjectResult)resourceExecutingContext.Result).Value.ToString());
```
Comment (llh-itm) [inline Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HubSpotValidationFilterTests.cs:107]: add !-operator to Result and Value to remove null warnings

Comment (llh-itm) [inline Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HubSpotValidationFilterTests.cs:107]: this should also be done to all the other tests in this file where this is done

File: Pelican.Infrastructure.HubSpot.Test/Mapping/CompanyResponseToClientTests.cs [hunk 4]
```diff
@@ -0,0 +1,321 @@
...
+		/// Act
+		Client result = response.ToClient();
+
+		/// Assert
+		Assert.Equal(SEGMENT, result.Segment);
+		Assert.Equal(NAME, result.Name);
+		Assert.Equal(LOCATION, result.OfficeLocation);
+		Assert.Equal(ID, result.HubSpotId);
+
+		Assert.Equal(0, result.Deals.Count);
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Infrastructure.HubSpot.Test/Mapping/CompanyResponseToClientTests.cs:38]: added

File: Pelican.Infrastructure.HubSpot.Test/Mapping/CompanyResponseToClientTests.cs [hunk 5]
```diff
@@ -0,0 +1,321 @@
...
+
+		Assert.Equal(0, result.ClientContacts.Count);
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Infrastructure.HubSpot.Test/Mapping/CompanyResponseToClientTests.cs:40]: added

[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Infrastructure.HubSpot.Test/Mapping/CompanyResponseToClientTests.cs:40]: added

File: Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HashComputer/HashComputerFactoryTests.cs [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+	[Fact]
+	public void CreateClientSecretHashComputer_ReturnsTypeSHA256HashComputer()
+	{
+		/// Act
+		var generator = _uut
+			.CreateClientSecretHashComputer(CLIENTSECRET);
+
+		/// Assert
+		Assert.Equal(
+		 typeof(ClientSecretHashComputer),
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HashComputer/HashComputerFactoryTests.cs:39]: indentation fixed 

clientsecret is private - test added to ClientSecretHashComputerTests to test exceptions in constructor

File: Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HashComputer/SHA256HashComputerTests.cs (hunk 1 above)
Comment (Nicolai-Bak) [reply Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HashComputer/SHA256HashComputerTests.cs:21]: fixed

File: Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HubSpotValidationFilterTests.cs (hunk 2 above)
Comment (Nicolai-Bak) [reply Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HubSpotValidationFilterTests.cs:98]: fixed

File: Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HubSpotValidationFilterTests.cs (hunk 3 above)
Comment (Nicolai-Bak) [reply Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HubSpotValidationFilterTests.cs:107]: fixed

Comment (Nicolai-Bak) [reply Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HubSpotValidationFilterTests.cs:107]: fixed

File: Pelican.Application/HubSpot/Commands/NewInstallation/NewInstallationCommandHandler.cs [hunk 7]
```diff
@@ -195,105 +195,3 @@ internal sealed class NewInstallationCommandHandler : ICommandHandler<NewInstall
 		return Result.Success();
 	}
 }
-
-
-//foreach (AccountManager accountManager in accountManagers)
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Application/HubSpot/Commands/NewInstallation/NewInstallationCommandHandler.cs:200]: removed

File: Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HashComputer/SHA256HashComputerTests.cs (hunk 1 above)
Comment (Nicolai-Bak) [reply Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HashComputer/SHA256HashComputerTests.cs:21]: test fails with code above - updated to: 
`string body = "[{" +
			"\"eventId\":1," +
			"\"subscriptionId\":12345," +
			"\"portalId\":62515," +
			"\"occurredAt\":1564113600000," +
			"\"subscriptionType\":\"contact.creation\"," +
			"\"attemptNumber\":0," +
			"\"objectId\":123," +
			"\"changeSource\":\"CRM\"," +
			"\"changeFlag\":\"NEW\"," +
			"\"appId\":54321}]";`

File: Pelican.Infrastructure.HubSpot.Test/Mapping/OwnerResponseToAccountManagerTests.cs [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+			Email = EMAIL,
+			UserId = USERID,
+
+		};
+
+		/// Act
+		AccountManager result = response.ToAccountManager();
+
+		/// Assert
+		Assert.Equal(ID, result.HubSpotId);
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Infrastructure.HubSpot.Test/Mapping/OwnerResponseToAccountManagerTests.cs:35]: Seems unnecessary to parse Guid to string just to parse it back again as type is already Guid.

File: Pelican.Infrastructure.HubSpot/Mapping/AccountManagers/OwnersResponseToAccountManagers.cs [hunk 9]
```diff
@@ -0,0 +1,20 @@
+﻿using Pelican.Domain.Entities;
+using Pelican.Infrastructure.HubSpot.Contracts.Responses.AccountManagers;
+
+namespace Pelican.Infrastructure.HubSpot.Mapping.AccountManagers;
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Infrastructure.HubSpot/Mapping/AccountManagers/OwnersResponseToAccountManagers.cs:4]: implicitly covered in hubspot service tests

File: Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HashGenerator/V1HashGeneratorTests.cs [hunk 10]
```diff
@@ -0,0 +1,91 @@
...
+	private readonly V1HashGenerator _uut;
+
+	private readonly HttpContext _httpContext;
+
+	public V1HashGeneratorTests()
+	{
+		_hashComputerFactoryMock = new();
+		_hashComputerMock = new();
+
+		_hashComputerFactoryMock
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Presentation.Api.Test/Utilities/HubSpotHookValidation/HashGenerator/V1HashGeneratorTests.cs:25]: to initialize unit under test the mock setup is needed

File: Pelican.Infrastructure.HubSpot.Test/Mapping/OwnerResponseToAccountManagerTests.cs (hunk 8 above)
[the remark this answers is not in the document]
Comment (AntonEsbjergIT-Minds) [reply Pelican.Infrastructure.HubSpot.Test/Mapping/OwnerResponseToAccountManagerTests.cs]: Fair

File: Pelican.Infrastructure.HubSpot.Test/Mapping/AccountManagers/OwnerResponseToAccountManagerTests.cs [hunk 11]
```diff
@@ -0,0 +1,90 @@
...
+	public void ToAccountManager_IdNull_ThrowException()
+	{
+		/// Arrange 
+		response.Id = null!;
+
+		/// Act
+		var result = Record.Exception(() => response.ToAccountManager());
+
+		/// Assert
+		Assert.NotNull(result);
```
[the remark this answers is not in the document]
Comment (Nicolai-Bak) [reply Pelican.Infrastructure.HubSpot.Test/Mapping/AccountManagers/OwnerResponseToAccountManagerTests.cs:35]: Asserts added

**Pull request merged**